test/utils/pickers-utils.tsx

```diff
@@ -402,62 +417,91 @@ export const buildFieldInteractions = <P extends {}>({
     });
   };
 
-  const selectSection: BuildFieldInteractionsResponse<P>['selectSection'] = (
-    input,
-    activeSectionIndex,
-  ) => {
-    const value = input.value.replace(':', '/');
-
-    // TODO: Improve this logic when we will be able to access state.sections from the outside
-    let clickPosition: number;
-    if (activeSectionIndex === 0) {
-      clickPosition = 0;
-    } else {
-      clickPosition =
-        (value.split('/', activeSectionIndex - 1).join('/').length +
-          value.split('/', activeSectionIndex).join('/').length) /
-        2;
+  const renderFromProps = (props: P, componentFamily: 'picker' | 'field' = 'field') => {
+    let fieldRef: React.RefObject<FieldRef<FieldSection>> = { current: null };
+
+    function WrappedComponent() {
+      fieldRef = React.useRef<FieldRef<FieldSection>>(null);
+      const allProps = {
+        ...props,
+      } as any;
+
+      if (componentFamily === 'field') {
+        allProps.unstableFieldRef = fieldRef;
+      } else {
+        if (!allProps.slotProps) {
+          allProps.slotProps = {};
+        }
+
+        if (!allProps.slotProps.field) {
+          allProps.slotProps.field = {};
+        }
+
+        const hasMultipleInputs =
+          // @ts-ignore
+          Component.render.name.includes('Range') &&
+          allProps.slots?.field?.fieldType !== 'single-input';
+        if (hasMultipleInputs) {
+          allProps.slotProps.field.unstableStartFieldRef = fieldRef;
+        } else {
+          allProps.slotProps.field.unstableFieldRef = fieldRef;
+        }
+      }
+
+      return <Component {...(allProps as P)} />;
     }
 
-    clickOnInput(input, clickPosition);
+    const result = render(<WrappedComponent />);
+
+    // Can't use `getTextbox` because we might have 2 input on multi input range fields.
```

Comment:
On the mobile version the role is different so we have 0 matches
BUT we never actually use the `input` on mobile, we just query it before it's inside a shared method.
I changed the selector to avoid adding complexity for the mobile :+1:

Reply:
Mhhh, true. It could be due to the fact that on mobile the field is `readonly` and in that case, RTL does not treat it as a valid `textbox` candidate. 🤔